101-vm-full-disk-encrypted-rhel/azuredeploy.json

```diff
@@ -0,0 +1,151 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "aadClientID": {
+      "metadata": {
+        "description": "Client ID of AAD app which has permissions to KeyVault"
+      },
+      "type": "string"
+    },
+    "aadClientSecret": {
+      "metadata": {
+        "description": "Client Secret of AAD app which has permissions to KeyVault"
+      },
+      "type": "securestring"
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the KeyVault to place the volume encryption key"
+      }
+    },
+    "keyVaultResourceGroup": {
+      "type": "string",
+      "metadata": {
+        "description": "Resource group of the KeyVault"
+      }
+    },
+    "vmName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the VM that will be created"
+      }
+    }
+  },
+  "variables": {
+    "createVmUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-vm-simple-rhel/azuredeploy.json",
+    "createVmDeploymentName": "[concat(uniquestring(parameters('vmName')), 'createVm')]",
+    "encryptVmUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-encrypt-running-linux-vm/azuredeploy.json",
+    "encryptVmDeploymentName": "[concat(uniquestring(parameters('vmName')), 'encryptVm')]",
+    "setupRaidUrl": "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-vm-full-disk-encrypted-rhel/setup_raid.sh",
```

Comment:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#samples-that-contain-extra-artifacts-custom-scripts-nested-templates-etc for handling artifacts that are part of the sample